autoload/lsp/diag.vim

```diff
@@ -456,16 +473,52 @@ export def GetDiagByPos(lspserver: dict<any>, bnr: number,
 enddef
 
 # Get all diagnostics from the LSP server for a particular line in a file
-export def GetDiagsByLine(lspserver: dict<any>, bnr: number, lnum: number): list<dict<any>>
-  if lspserver.diagsMap->has_key(bnr)
-    var diagsbyLnum = lspserver.diagsMap[bnr].diagnosticsByLnum
-    if diagsbyLnum->has_key(lnum)
-      return diagsbyLnum[lnum]->sort((a, b) => {
-	  return a.range.start.character - b.range.start.character
-	})
+export def GetDiagsByLine(bnr: number, lnum: number, lspserver: dict<any> = null_dict): list<dict<any>>
+  if !diagsMap->has_key(bnr)
+    return []
+  endif
+
+  var diagsByLnum: list<dict<any>> = []
+
+  if lspserver != null_dict
+    if !diagsMap[bnr]->has_key(lspserver.id)
+      return []
     endif
+    diagsByLnum = diagsMap[bnr][lspserver.id]
+  else
+    for lspDiags in diagsMap[bnr]->values()
+      if !lspDiags.diagnosticsByLnum->has_key(lnum)
+        continue
+      endif
+
+      diagsByLnum->extend(lspDiags.diagnosticsByLnum[lnum])
+    endfor
+  endif
+
+  return diagsByLnum->sort((a, b) => {
+    return a.range.start.character - b.range.start.character
+  })
+enddef
+
+# Get all diagnostics from the LSP server for a particular line in a file
+export def GetSortedDiags(bnr: number): list<dict<any>>
+  if !diagsMap->has_key(bnr)
+    return []
   endif
-  return []
+
+  # TODO Optimize this function
+
+  var allDiagnostics: list<dict<any>> = []
+  for lspDiags in diagsMap[bnr]->values()
+
+    if lspDiags.sortedDiagnostics->empty()
+      continue
+    endif
+
+    allDiagnostics->extend(lspDiags.sortedDiagnostics)
+  endfor
+
+  return SortDiags(allDiagnostics)
```

Comment:
Currently the diagnostics are sorted a lot of times, first when they arrive, and later when each list of LSP diagnostics are being merged into one.